### PR TITLE
fix: [M02] incorrect math in slashing library comments

### PR DESCRIPTION
**Pull request by md0x**
**Motivation**

```
The formula uses basic interest instead of compound interest and assumes that a 20% increase will be
negated with a subsequent 20% loss. However, the correct formula to counteract a 20% APY
over 120 votes would be
1 - (1 / 1.2)**(1/120)
```

**Summary**

- Updated slashing calculation formula to take slashing compound into account
- Added a note to remind people to check for a formula inequality when updating the library

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [x]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

**Review by chrismaree — APPROVED** [review]

File: packages/core/contracts/oracle/implementation/SlashingLibrary.sol [hunk 1]
```diff
@@ -15,7 +15,9 @@ contract SlashingLibrary {
         uint256 totalCorrectVotes
     ) public pure returns (uint256) {
         // This number is equal to the slash amount needed to cancel an APY of 20%
-        // if 10 votes are cast each month for a year. 0.2/(10*12)= ~0.0016
+        // if 10 votes are cast each month for a year.  1 - (1 / 1.2)**(1/120) = ~0.0016
```
Comment (mrice32) [inline packages/core/contracts/oracle/implementation/SlashingLibrary.sol:18]: The result of this function is closer to 0.0015 than 0.0016, should we change the constant or something?

**Review by mrice32 — COMMENTED** [review]
Note: if this is just an issue in the comments, I think this should not be considered a medium severity issue. At most, it should be considered L, or potentially an N. We should request that this be reclassified IMO.

The fix looks good! The updated math changes the result enough that the number now no longer seems to be the correct rounded value. Should we change it?

**Review by mrice32 — APPROVED** [review]

File: packages/core/contracts/oracle/implementation/SlashingLibrary.sol (hunk 1 above)
Comment (md0x) [reply packages/core/contracts/oracle/implementation/SlashingLibrary.sol:18]: We have 13 tests that rely on this `0.0016` to be correct: the values are hardcoded to showcase the maths; we can change it, but I'm not sure what the added value is if we plan to replace it with a simpler value later on, such as `0.001` 

I am happy to change them if you think we should change to 0.0015. If so, maybe I should remove the hardcoded values from the tests? cc @chrismaree

Comment (chrismaree) [reply packages/core/contracts/oracle/implementation/SlashingLibrary.sol:18]: we should not change this. leave it at 0.0016, the time taken to update all the tests and the comments is not worth it. this magic number was chosen as something approximately useful or close to what we'd like. it was not meant to be accurate or precise; we'll likely start at something like `0.001` when we deploy the contracts so it's really easy to reason about

(note when I set up the tests I intentionally did not choose `0.001` as using perfect round numbers can obfuscate complexity in computations as you end up scaling things up or down by a factor which can cause issues when reasoning about things).

**Pull request merged**